be/src/storage/lake/vertical_compaction_task.cpp

```diff
@@ -113,7 +113,12 @@ StatusOr<int32_t> VerticalCompactionTask::calculate_chunk_size_for_column_group(
         ASSIGN_OR_RETURN(auto segments, rowset->segments(false, true));
         for (auto& segment : segments) {
             for (auto column_index : column_group) {
-                const auto* column_reader = segment->column(column_index);
+                auto uid = _tablet_schema->column(column_index).unique_id();
+                if (!segment->is_valid_column(uid)) {
+                    continue;
+                }
+
+                const auto* column_reader = segment->column_with_uid(uid);
```

Comment:
I think we should probably modify the implementation of `Segment::column()` to return nullptr when a column doesn't exist in the segment file, instead of throwing an exception that causes a crash. this would avoid more similar problems like the one here.

Reply:
Fixed